lib/responder.js

```diff
@@ -248,17 +276,18 @@ MultipartDispatcher.prototype.promise = function dispatchMultipartPromise(
   var objectQueue     = new FifoQueue(2);
   var partReaderQueue = new FifoQueue(3);
 
-  var metadataBuffer  = null;
-
   var parsingParts = 0;
   var parsedParts  = 0;
 
   var hasParsed = false;
   var hasEnded  = false;
```

Comment:
Is there a need for hasEnded (and the end listener) in the Promise case?

Reply:
We currently have a responseEndListener that sets hasEnded in the multipart promise case. Are saying when that happens, make sure to drain Dicer in case that stream is currently paused?

Reply:
Sorry for the inclarity.  I'm saying that I believe there's never any point in firing the drain event -- not on Dicer or any other WritableStream.  My reading of https://nodejs.org/dist/latest-v8.x/docs/api/stream.html#stream_event_drain is that the drain event is a notification fired by the WriteableStream and not a request to the WriteableStream. In other words, the existing code was grossly mistaken in firing this event.

Reply:
OK, that's how I understand it, too. I will remove the firing of the "drain" event in the chunked case, I'd overlooked that. And since we do a direct pipe of data in the chunked case, and pipe handles backpressure, we should be OK there.